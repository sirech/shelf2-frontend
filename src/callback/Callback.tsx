import React, { useEffect } from 'react'

import { Navigate } from 'react-router-dom'
import { useAppDispatch, useAppSelector } from 'hooks'

import { actions, authenticatedSelector } from 'state/login'

const redirectIfLoggedIn = (authenticated: boolean) => {
  if (authenticated) {
    return <Navigate replace to="/" />
  }
}

const Callback = () => {
  const authenticated = useAppSelector(authenticatedSelector)
  const dispatch = useAppDispatch()

  useEffect(() => {
    if (!authenticated) {
      dispatch(actions.login())
    }
  }, [authenticated, dispatch])

  return <div>{redirectIfLoggedIn(authenticated)}</div>
}

export default Callback
