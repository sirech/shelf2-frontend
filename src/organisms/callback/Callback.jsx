// @flow

import React from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { Redirect } from 'react-router-dom'

import { actionPicker } from 'state'
import { actions, authenticatedSelector } from 'state/login'

type Props = {
  authenticated: boolean,
  login: () => undefined,
}

class Callback extends React.Component {
  static defaultProps: Props
  props: Props

  redirectIfLoggedIn() {
    if (this.props.authenticated) {
      return <Redirect to="/" />
    }
  }

  componentDidMount() {
    if (!this.props.authenticated) {
      this.props.login()
    }
  }

  render() {
    return <div>{this.redirectIfLoggedIn()}</div>
  }
}

Callback.defaultProps = {
  authenticated: false,
  login: _ => undefined,
}

export default connect(
  (state, props) =>
    createStructuredSelector({
      authenticated: authenticatedSelector,
    })(state),
  actionPicker(['login'])(actions)
)(Callback)
