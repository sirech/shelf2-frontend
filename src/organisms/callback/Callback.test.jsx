import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Callback } from './Callback'

describe('components', () => {
  describe('Callback', () => {
    it('does not blow up', () => {
      const component = shallow(<Callback />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('renders a redirect if authenticated', () => {
      const component = shallow(<Callback authenticated />)
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
