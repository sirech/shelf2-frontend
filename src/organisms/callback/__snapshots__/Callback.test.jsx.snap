// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Callback does not blow up 1`] = `<div />`;

exports[`components Callback renders a redirect if authenticated 1`] = `
<div>
  <Redirect
    to="/"
  />
</div>
`;
